src/commands/build.ts

```diff
@@ -85,36 +83,58 @@ export class BuildCommand implements Command {
     },
   ];
 
-  run(options: CommandOptions, config: ProjectConfig): Promise<any> {
+  async run(options: CommandOptions, config: ProjectConfig): Promise<any> {
     // Defer dependency loading until this specific command is run
-    const build = require('../build/build').build;
+    let build = require('../build/build').build;
+    const hasCLIArgumentsPassed = this.args.some(arg => {
+      return typeof options[arg.name] !== 'undefined';
+    });
 
-    let buildOptions: BuildOptions = {
-      addServiceWorker: options['add-service-worker'],
-      swPrecacheConfig: options['sw-precache-config'],
-      insertPrefetchLinks: options['insert-prefetch-links'],
-      bundle: options['bundle'],
-      html: {
-        minify: !!options['html-minify'],
-      },
-      css: {
-        minify: !!options['css-minify'],
-      },
-      js: {
-        minify: !!options['js-minify'],
-        compile: !!options['js-compile'],
-      },
-    };
+    // Support passing a custom build function via options.env
+    if (options['env'] && options['env'].build) {
+      logger.debug('build function passed in options, using that for build');
+      build = options['env'].build;
+    }
 
-    logger.debug('building with options', buildOptions);
+    const mainBuildDirectory = 'build/';
+    logger.info(`Clearing ${mainBuildDirectory} directory...`);
+    await del([mainBuildDirectory]);
 
-    if (options['env'] && options['env'].build) {
-      logger.debug('env.build() found in options');
-      logger.debug('building via env.build()...');
-      return options['env'].build(buildOptions, config);
+    // If any the build command flags were passed as CLI arguments, generate
+    // a single build based on those flags alone.
+    if (hasCLIArgumentsPassed) {
```

Comment:
Ok, to iterate on that, I don't think it's going to be very ergonomic to use an index-based build name. I'd rather just require names, with the possible exception of only supporting an unnamed build if it's the only build.